### PR TITLE
Support use K8s etcd(configmap) as datastore

### LINKED ISSUE #130: velacp should support use K8s etcd(configmap) as datastore
We should not rely on mongodb as a must.

### PR DESCRIPTION
**Pull request by wangyuan249**
fix #130 

- [x]  Change to store cluster rpc struct info to config map.
- [x]  Change to store application rpc struct info to config map.
- [x]  Change to store other object which we didn't define in kubernetes CRD in config map.

### CODE REVIEW

**Comment by wangyuan249** [comment]
First version，
1. Shall we create a specific namespace when the echo httpserver is starting for better management of vela ui resource, eg: "velaui-system" ?
2. The name for each configmap should be same with the resource name or add a suffix like "appinstance1-schema"
3. After all planned fucntion has been finished,  I would do some code simplification and abstract the shared part.

File: go.mod [hunk 1]
```diff
@@ -15,6 +15,7 @@ require (
 	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
 	github.com/oam-dev/kubevela v1.0.5
 	github.com/pkg/errors v0.9.1
+	github.com/prometheus/common v0.10.0
```
Comment (hongchaodeng) [inline go.mod:18]: What's this dependency for?

File: pkg/rest/services/cluster.go [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
 
 	"github.com/oam-dev/velacp/pkg/datastore/storeadapter"
 	"github.com/oam-dev/velacp/pkg/proto/model"
 	"github.com/oam-dev/velacp/pkg/rest/apis"
+	initClient "github.com/oam-dev/velacp/pkg/rest/client"
 )
 
 type ClusterService struct {
-	store storeadapter.ClusterStore
+	store     storeadapter.ClusterStore
```
Comment (hongchaodeng) [inline pkg/rest/services/cluster.go:28]: This store is not used anymore. Right? Why not delete it?

**Comment by wangyuan249** [comment]
1. Cause there are many changes involved,  At present, we still lack tests for each condition and situation in different request and response. I only tested the most common scenarios.
2. I will add a description file to introduce how config map store the information and for different types config map we would add a common label for better list and management.
3. For better error report and error location, I would continue to improve the error-throw out mechanism, include http status code, error code definition and error message.

**Review by hongchaodeng — APPROVED** [review]
LGTM

File: pkg/rest/rest_server.go [hunk 3]
```diff
@@ -71,32 +86,52 @@ func (s *restServer) Run(ctx context.Context) error {
...
-	s.server.GET("/api/catalogs/:catalogName/capabilities", catalogService.GetCapabilities)
-	s.server.POST("/api/catalogs/:catalogName/sync", catalogService.SyncCatalog)
 
-	clusterStore := storeadapter.NewClusterStore(s.ds)
-	clusterService := services.NewClusterService(clusterStore)
+	// cluster
+	clusterService, err := services.NewClusterService()
+	if err != nil {
+		log.Logger.Errorf(err.Error())
+	}
```
Comment (wonderflow) [inline pkg/rest/rest_server.go:134]: 遇到error确定还要继续？

**Review by wonderflow — CHANGES_REQUESTED** [review]

File: pkg/rest/rest_server.go (hunk 3 above)
Comment (wangyuan249) [reply pkg/rest/rest_server.go:134]: fixed

**Comment by hongchaodeng** [comment]
@wangyuan249 Please also fix the CI :)

**Comment by wangyuan249** [comment]
fix CI
image
image

**Comment by wangyuan249** [comment]
image

**Comment by hongchaodeng** [comment]
Awesome. Thanks!

**Pull request merged**